### PR TITLE
PSR-574 post transformation of landOrProperty disposal data

### PR DESCRIPTION
**Pull request by cyeniceri**

### CODE REVIEW

File: app/services/PsrSubmissionService.scala [hunk 1]
```diff
@@ -60,11 +62,13 @@ class PsrSubmissionService @Inject()(
           minimalRequiredSubmission = minimalRequiredSubmission,
           checkReturnDates = checkReturnDates,
           loans = Option.when(schemeHadLoans)(Loans(schemeHadLoans, loanTransactionsTransformer.transformToEtmp(srn))),
-          assets = Option.when(landOrPropertyHeld || moneyWasBorrowed)(
+          assets = Option.when(landOrPropertyHeld || moneyWasBorrowed || disposeAnyLandOrProperty)(
```
Comment (kberec) [inline app/services/PsrSubmissionService.scala:65]: What happens in a scenario when there are properties (or borrowings) but no disposals?

Comment (cyeniceri) [reply app/services/PsrSubmissionService.scala:65]:
```
 optDisposedPropertyTransaction =
                      Option.when(disposeAnyLandOrProperty)(buildOptDisposedPropertyTransactions(srn, index))
```

Then optDisposedPropertyTransaction will be None and we'll send nothing to BE

Comment (kberec) [reply app/services/PsrSubmissionService.scala:65]: Is that right though? We should send landOrProperty (and borrowing) even if there are no disposals.

Comment (cyeniceri) [reply app/services/PsrSubmissionService.scala:65]: yes I mean, we are not going to send any DisposedPropertyTransaction (if false) but landOrProperty (and borrowing).

**Review by kberec — APPROVED** [review]

**Pull request merged**